### DISCUSSION TITLE
[JS101] 關於一、二、三：三元運算子（Ternary）

### DISCUSSION THREAD

**Discussion by ming320003** [Lidemy/forum#42] [前端相關問題]
不好意思我又有問題想要請教。
在這段影片的 0:20 ，老師寫了 var isPass = false 。請問這句的用意是什麼呢？
就語法上的意思應該是宣告變數 isPass 然後把 false值給它。但是我看不太懂為什麼要這樣做 ( 在一開始就給它一個 false 值 )，難道是因為程式執行到下面的 if (score) ... 階段，如果分數大於等於 60 ，isPass 會變成 true ，反正會取代 false 所以沒關係嗎？但是即使沒有關係，又為什麼是要寫 false 呢？ 
謝謝老師

**Comment by ming320003** [comment]
我覺得我好像想通了，應該就是和 var score = 60 一樣的道理，就是隨便給的，var isPass = false 這裡不論用 false 還是 true 對最後的答案好像都不影響。不知道我最後想的對不對？

**Comment by aszx87410** [comment]
之後提問可以把相關程式碼截圖放上來一下，否則單憑這個提問不會知道完整的情境XD

我猜程式碼可能是類似這樣：

```
var isPass = false
if (score > 60) {
  isPass = true
}
```

第一行就是給 isPass 預設值，不是隨便給的，讓他預設值是 false 代表不通過，底下判斷 score>60 才讓他變成 true 代表通過，所以如果沒有走到 if(score>60) 那裡面，isPass 就會是 false

**Reply by ming320003** [reply]
原來是這樣，真是有趣，謝謝老師的解答。
補原本的 code
var score = 60
var isPass = false
if (score >= 60) {
  isPass = true
} else {
  isPass = false
}
